src/cpp/function.cpp

```diff
@@ -98,7 +114,10 @@ sol::object Function::loadFunction(lua_State* state) {
         sol::stack::push(state, obj);
         lua_setupvalue(state, -2, i + 1);
     }
-    return sol::stack::pop<sol::function>(state);
+    const auto obj = sol::stack::pop<sol::object>(state);
+    cache::put(state, *this, obj);
```

Comment:
Here should be one invocation of `put`.

Reply:
Currently, we have 2 different `put` functions, so here we have two invocation. If we will replace them by single one, there will be only one call.